araproc/analysis/standard_reco.py

```diff
@@ -1247,20 +1247,26 @@ def get_csw(
             # Trim this waveform's length to match the CSW length
             if len(times) > len(csw_times):
                 trim_ammount = len(times) - len(csw_times)
-                if (times[0]<csw_times[0]) and (times[-1]<=csw_times[-1]): 
-                    # If this wf has an earlier start time and a later or equal 
-                    #   end time, trim from front
+                if (
+                    ( times[0] - csw_times[0] < 0 ) # this wf has an earlier start time than the CSW
```

Comment:
I don't think so? The way I explained it to myself is that everything written with respect to 0 was a proxy for an inequality that doesn't consider the case of equality like < and > (like `times[0] - csw_times[0] < 0` is a proxy for `times[0] < csw_times[0]`) and everything written with respect to dt/2 is a proxy for an inequality that includes the case of equality like >= and <= (like `times[0] - csw_times[0] > dt/2` is a proxy for `times[0] >= csw_times[0]`)

Reply:
Okay I see what you mean. Maybe I'm confused how you decide if it's the front or the back of the array that needs to be trimmed then?

Reply:
Maybe I should add to the comments but you had suggested that if a channel's waveform needs to be trimmed  to match the length of the CSW then we should trim off the datapoints corresponding to times that don't overlap with the CSW. For example, if the CSW has a time array of [4, 5, 6, 7] and channel 1 has a waveform with the time array [2, 3, 4, 5, 6] then we'd want to trim off 1 data point to get the length of channel 1's waveform to match the length of the CSW. We'd want to remove the data point from channel 1's waveform at time t=2 (from the front) instead of time=6 (from the back) since time=6 is within the time range of the CSW.

Getting the math right took a lot of whiteboard drawing so I can add some examples to the comments if it would help with understanding this part of the code

Reply:
I have another idea for how to do this part that doesn't include trimming and a huge if/else block that I can run by you if you'd like but it includes quite a bit of reworking and rethinking

Reply:
Ah okay I think that makes sense. Maybe the comment on the next line threw me off. I think it should say the this wf has an earlier or equal end time? If that's right, then maybe just a quick double-check of the comments is all that is needed.

Reply:
You're totally right, I switched that comment around. Fixed!